src/main/java/com/minecolonies/coremod/colony/buildings/registry/BuildingDataManager.java

```diff
@@ -59,7 +62,12 @@ public IBuilding createFrom(final IColony colony, final BlockPos position, final
             return null;
         }
 
-        return entry.getBuildingProducer().apply(colony, position);
+        final IBuilding building = entry.getBuildingProducer().apply(colony, position);
```

Comment:
could probably make this easier with an entry.apply() function and do the necessary logic within right?